### DISCUSSION TITLE
AwoX PIR (AwoX 99106) recognised as AwoX BULB 33951/33948

### DISCUSSION THREAD

**Discussion by MaurizioITA** [Koenkk/zigbee2mqtt#13817] [Devices]
Hello,
I'm adding a couple of "AWOX Zigbee PIR SENSOR 99106" to Zigbee2MQTT that are recognised as "AWOX bulb 33951/33948"!
This is the product: https://www.eglo.com/uk/accessory-connect-z-sensor-99106.html
How can I fix tis issue?

![immagine](https://user-images.githubusercontent.com/91998029/188155672-0b2b4367-3a3c-4f57-a28e-21c3f267dc3b.png)

**Comment by Koenkk** [comment]
Did you already upgrade to z2m 1.27.2? If it still appears there, please provide me the data/database.db entry of this device.

**Reply by MaurizioITA** [reply]
Hi Koenkk,
Yes the version is 1.27.2
Here it is the DB: [database.txt](https://github.com/Koenkk/zigbee2mqtt/files/9500685/database.txt)

Inside you will find a lot of Awox components:
99099 (0xa4c13834a5289430) remote control, detected as 33952 remote
12251 GU10 (0xa4c138228cfb215f), detected as 33943/33944/33946 bulb
They works but are not formally correct - aesthetics

Obviously also the 99106 PIR (0xa4c1387f3cdcdf1f) detected as 33951/33948 bulb that doesn't work

**Reply by Koenkk** [reply]
The following external converter should make the device correctly detected. However since this device is not support yet, you have to add support for it first (filling `toZigbee`/`fromZigbee`/`exposes`). A tutorial can be found here: https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#instructions



```js
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    fingerprint: [
        {
            type: 'EndDevice', manufacturerName: 'AwoX', modelID: 'TLSR82xx', powerSource: 'Battery', endpoints: [
                {ID: 1, profileID: 260, deviceID: 263, inputClusters: [0, 3, 4, 4096], outputClusters: [0, 3, 4, 5, 6, 8, 768, 4096]},
                {ID: 3, profileID: 4751, deviceID: 263, inputClusters: [65360, 65361], outputClusters: [65360, 65361]},
                {ID: 4, profileID: 260, deviceID: 263, inputClusters: [0,3,4,1024,1030,4096,32], outputClusters: [0,3,4,1024,1030,4096,18778,16967]},
            ],
        },
    ],
    model: '99106',
    vendor: 'AwoX',
    description: 'PIR sensor',
    exposes: [],
    toZigbee: [],
    fromZigbee: [],
};

module.exports = definition;

```

- save this as file next to `configuration.yaml` as `ext_converter.js`
- add it to `configuration.yaml`:
```yaml
external_converters:
  - ext_converter.js
```
- start z2m

**Reply by MaurizioITA** [reply]
Tks Koenkk,
now it is detected but doesn't work yet.
I'll try to configure it

**Reply by MaurizioITA** [reply]
Hi Koenkk,
I've tried to configure the sensors a lot of times, but with no results.
I've recived every type of messages like this:

![immagine](https://user-images.githubusercontent.com/91998029/215164984-04bc1307-28d3-4edf-ab81-bf44e8c88f49.png)

the configuration of the sensor are:

const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    fingerprint: [
        {
            type: 'EndDevice', manufacturerName: 'AwoX', modelID: 'TLSR82xx', powerSource: 'Battery', endpoints: [
                {ID: 1, profileID: 260, deviceID: 263, inputClusters: [0, 3, 4, 4096], outputClusters: [0, 3, 4, 5, 6, 8, 768, 4096]},
                {ID: 3, profileID: 4751, deviceID: 263, inputClusters: [65360, 65361], outputClusters: [65360, 65361]},
                {ID: 4, profileID: 260, deviceID: 263, inputClusters: [0,3,4,1024,1030,4096,32], outputClusters: [0,3,4,1024,1030,4096,18778,16967]},
            ],
        },
    ],
    model: '99106',
    vendor: 'AwoX',
    description: 'PIR sensor',
    exposes: [e.occupancy()],
    toZigbee: [],
    fromZigbee: [fz.command_on, fz.command_off, fz.ias_occupancy_alarm_1, fz.ias_occupancy_alarm_2, fz.occupancy],
    

};

    
module.exports = definition;

How can i fix?
